asb-ballerina/messageReceiver.bal

```diff
@@ -21,26 +21,42 @@ import ballerina/jballerina.java as java;
+    private string topicName;
+    private string receiveMode;
     final handle receiverHandle;
 
     # Initializes the connector. During initialization you can pass the [Shared Access Signature (SAS) authentication credentials](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas)
-    # Create an [Azure account](https://docs.microsoft.com/en-us/learn/modules/create-an-azure-account/) and 
-    # obtain tokens following [this guide](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal#get-the-connection-string). 
+    # Create an [Azure account](https://docs.microsoft.com/en-us/learn/modules/create-an-azure-account/) and
+    # obtain tokens following [this guide](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal#get-the-connection-string).
     # Configure the connection string to have the [required permission](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas).
-    # 
-    # + connectionString - Connection String of Azure service bus
-    # + entityPath - Name or path of the entity (e.g : Queue name, Subscription path)
+    #
+    # + connectionString - Connection String of Azure service bus  
+    # + queueName - Queue name  
+    # + topicName - Topic name  
+    # + subscriptionName - Subscription name  
     # + receiveMode - Receive mode as PEEKLOCK or RECEIVEANDDELETE (default : PEEKLOCK)
-    public isolated function init(@display {label: "Connection String"} string connectionString, @display {label: "Entity Path"} string entityPath, 
-                                  @display {label: "Receive Mode"} string receiveMode = PEEKLOCK) returns error? {
-        self.connectionString = connectionString;
-        self.entityPath = entityPath;
-        self.receiveMode = receiveMode;
+    # + return - Return Value Description
+    public isolated function init(ASBServiceReceiverConfig config) returns error? {
+        self.connectionString = config.connectionString;
+        if config.entityType == "queue" {
+            QueueConfig queueConfig = check config.entityConfig.ensureType(QueueConfig);
+            self.queueName = queueConfig.QueueName;
+            self.subscriptionName = EMPTY_STRING;
+            self.topicName = EMPTY_STRING;
+        } else {
+            TopicSubsConfig topicSubsConfig = check config.entityConfig.ensureType(TopicSubsConfig);
+            self.topicName = topicSubsConfig.topicName;
+            self.subscriptionName = topicSubsConfig.subscriptionName;
+            self.queueName = EMPTY_STRING;
+        }
+        
+        self.receiveMode = config.receiveMode;
         self.receiverHandle = check initMessageReceiver(java:fromString(self.connectionString), 
-        java:fromString(self.entityPath), java:fromString(self.receiveMode));
+        java:fromString(self.queueName),java:fromString(self.topicName), java:fromString(self.subscriptionName), 
```

Comment:
Why do we get `queueName ` , `topicName ` and `subscriptionName ` like this?

Reply:
Here, these values are sent to ASB receiver constructor and its parameters are Java strings. (java.lang.String). Therefore, we take ballerina strings and converts them to java type string by using java:fromString. then they are passed using ballerina handles. AFAIK and I tested removing these java type conversion, it is not possible to initialize the java native receiver class. Please refer here https://ballerina.io/learn/java-interoperability-guide/ballerina-ffi/#deal-with-overloaded-constructors